### DISCUSSION TITLE
Testing

### DISCUSSION THREAD

**Discussion by matt-42** [matt-42/lithium#50] [Ideas]
@cong1920 , let's continue the discution on testing here.

I found doctest https://github.com/onqtam/doctest which is a rewrite on catch, providing similar features but faster compile time: what do you think of this one ?

**Comment by cong1920** [comment]
I did give DocTest a shot with some of my Leetcode problem solving code. Yes, compiling feels definitely faster than Catch. And the test output is as same clean as Catch. Glad you you found it! I will try some Lithium test code with DocTest next. 

What is your idea on how to add this DocTest into Lithium's repository if building does not become significant slow? Add as a git submodule, or just directly download its release package from https://github.com/onqtam/doctest/releases before first time Lithium code local compile. I personally prefer git submodule :)

**Comment by jmafc** [comment]
@matt-42 I commented on `doctest` on the other thread. I give it a thumbs up.
